frontend/src/features/stock-view/components/StockUpdateForm.tsx

```diff
@@ -0,0 +1,216 @@
+                    <div className="flex items-center justify-end pt-7">
+                        <Checkbox
+                            id="isBlocked"
+                            checked={newStock?.isBlocked ?? false}
+                            onChange={(event) => dispatch({ type: 'isBlocked', payload: event.target.checked })}
+                        />
+                        <label htmlFor="isBlocked"> is Blocked </label>
+                    </div>
+                </div>
+                <div className="w-[32rem]">
+                    <Autocomplete
+                        id="site"
+                        value={newStock?.stockLocationBpns ?? null}
+                        options={sites?.map((site) => site.bpns) ?? []}
+                        renderInput={(params) => <Input {...params} label="Stock Location BPNS*" placeholder="Select a Site" />}
+                        onChange={(_, value) => dispatch({ type: 'stockLocationBpns', payload: value ?? null })}
+                    />
+                    <Autocomplete
+                        id="site"
+                        value={newStock?.stockLocationBpna ?? null}
+                        options={sites?.find((site) => site.bpns === newStock?.stockLocationBpns)?.addresses.map((addr) => addr.bpna) ?? []}
+                        renderInput={(params) => <Input {...params} label="Stock Location BPNA*" placeholder="Select an Address" />}
+                        onChange={(_, value) => dispatch({ type: 'stockLocationBpna', payload: value ?? null })}
+                    />
+                    <div className="grid grid-cols-2 gap-2">
+                        <Input
+                            id="customer-order-number"
+                            label="Customer Order Number"
+                            type="text"
+                            value={newStock?.customerOrderNumber}
+                            onChange={(event) => dispatch({ type: 'customerOrderNumber', payload: event.target.value })}
+                        />
+                        <Input
+                            id="customer-order-position"
+                            label="Customer Order Position"
+                            type="text"
+                            value={newStock?.customerOrderPositionNumber}
+                            onChange={(event) => dispatch({ type: 'customerOrderPositionNumber', payload: event.target.value })}
+                        />
+                    </div>
+                    <Input
+                        id="supplier-order-number"
+                        label="Supplier Order Number"
+                        type="text"
+                        value={newStock?.supplierOrderNumber}
+                        onChange={(event) => dispatch({ type: 'supplierOrderNumber', payload: event.target.value })}
+                    />
+                </div>
+            </div>
+            <div className="mt-7 mx-auto w-48">
+                <LoadingButton
```

Comment:
Functionality is not working. neither add nor update

Reply:
update is working, add not. Also order position reference is not considered.

a stock is new, if no other stock exists with same
- material / product
- unit of measure (quantity)
- allocated to partner
- is blocked
- location (BPNS, PBNA)
- Order Position Reference (all three attributes)

Reply:
I investigated the add issue a little deeper: Seems like the issue comes from order Position References. If empty, they are are expected as null.